README.md

```diff
@@ -44,10 +45,162 @@ thrift_files: Mix.Utils.extract_files(["thrift"], [:thrift])
+`timeout`        | A positive integer | The amount of time in milliseconds the Client's GenServer waits for a reply. After this, the GenServer will exit with `{:error, :timeout}`.
+
+
+### Example of using options
+
+```elixir
+alias Thrift.Test.UserService.Clients.Binary.Framed, as: Client
+{:ok, client} = UserService.Clients.Binary.Framed.start_link("localhost", 2345, 
+                tcp_opts: [backoff_calculator: fn(retry_count) -> retry_count * 1000], gen_server_opts: [timeout: 10_000])
+                
+```
+In the above example, the client will use a (very) conservative backoff calculator that waits an additional second each time the client retries. In other words, the first retry will wait 1 second, the second will wait 2 seconds, and the third will wait 3. It sets the GenServer timeout to be ten seconds, which means the remote side can take its time to reply. 
+
+
+## Using The Server
+Creating a thrift server is slightly more involved than creating the client, because you need to create a module to handle the work. Fortunately, a `Behaviour` for this module, complete with correct success typing, is generated for you. To implement this behaviour, use the `@behaviour` module attribute. The compiler will now inform you about any missed functions.
+
+```elixir
+defmodule UserServiceHandler do 
+  @behaviour Thrift.Test.UserService.Handler.Behaviour
+  
+  def ping, do: true 
+  
+  def get_user_by_id(user_id) do 
+	case Backend.find_user_by_id(user_id) do 
+		{:ok, user} -> 
+			user 
+		{:error, _} -> 
+			raise Thrift.Test.UserNotFound.exception message: "could not find user with id #{user_id}"
+    end
+  end
+  
+  def delete_user(user_id) do 
+    case Backend.delete_user(user_id) do 
+	   :ok -> true 
+	   _   -> false
+  end
+end
+```
+
+
+After you have your handler module, you use it to build your server. 
+
+```elixir
+{:ok, server_pid} = Thrift.Test.UserService.Servers.Binary.Framed.start_link(UserServiceHandler, 2345, [])
+```
+
+...and your server is up and running. RPC calls made to the server will be delegated to your handler module. 
+
+Like the client, the server takes several options. They are:
+
+Name           |  Type | Description 
+---------------|-------|-------------
+worker_count   | positive integer | The number of acceptor workers available to take requests
+name  | atom | (Optional) The name of the server. The server's pid becomes registered to this name. If not specified, the handler module's name is used.
+max_restarts | non negative integer | The number of times to restart (see the next option)
+max_seconds  | non negative integer | The number of seconds. This is used by the suprvisor to determine when to crash. If a server restarts `max_restarts` times in `max_seconds` then the supervisor crashes.
+
+The server defines a Supervisor, which can be added to your application's supervision tree. Don't forget to use the `supervisor` function rather than the `worker` function. 
```

Comment:
"Use the `supervisor` function rather than the `worker` function to do this."?   The "Don't forget" part is a little confusing to me.